pages/reference/OS/configuration.md

```diff
@@ -0,0 +1,179 @@
+
+`cat rulefilename | jq -sR .`
+
+For example:
+
+```shell
+root@resin:/etc/udev/rules.d# cat 64.rules | jq -sR .
+"ACTION!=\"add|change\", GOTO=\"modeswitch_rules_end\"\nKERNEL==\"ttyACM*\", ATTRS{idVendor}==\"1546\", ATTRS{idProduct}==\"1146\", TAG+=\"systemd\", ENV{SYSTEMD_WANTS}=\"u-blox-switch@'%E{DEVNAME}'.service\"\nLBEL=\"modeswitch_rules_end\"\n"
+```
+
+The following example contains two custom udev rules that will create `/etc/udev/rules.d/56.rules` and `/etc/udev/rules.d/64.rules`. The first time rules are added/modified, udevd will reload the rules and re-trigger.
+
+```json
+"os": {
+  "udevRules": {
+    "56": "ENV{ID_FS_LABEL_ENC}==\"resin-root*\", IMPORT{program}=\"resin_update_state_probe $devnode\", SYMLINK+=\"disk/by-state/$env{RESIN_UPDATE_STATE}\"",
+    "64" : "ACTION!=\"add|change\", GOTO=\"modeswitch_rules_end\"\nKERNEL==\"ttyACM*\", ATTRS{idVendor}==\"1546\", ATTRS{idProduct}==\"1146\", TAG+=\"systemd\", ENV{SYSTEMD_WANTS}=\"u-blox-switch@'%E{DEVNAME}'.service\"\nLBEL=\"modeswitch_rules_end\"\n"
+  }
+}
+```
+
+#### sshKeys
+
+(Array) An array of strings containing a list of public SSH keys that will be used by the SSH server for authentication.
+
+```json
+"os": {
+  "sshKeys": [
+    "ssh-rsa AAAAB3Nza...M2JB balena@macbook-pro",
+    "ssh-rsa AAAAB3Nza...nFTQ balena@zenbook"
+  ]
+}
+```
+
+## Managing config.json on a provisioned device
+
+Once the device has been provisioned, the contents of `config.json` are copied to the device boot partition at `/mnt/boot/config.json`. You can view the contents of this file on a provisioned device with the following command:
+
+```shell
+balena ssh <uuid>
+cat /mnt/boot/config.json | jq '.'
+```
+
+Special care should be taken when modifying any values of `config.json` on a provisioned device. This file is also used by the device supervisor, and mistakes to this file can leave a device inaccessible. There are various ways of safely modifying the values of `config.json`:
+
+* Using the supervisor API, you can update the device [hostname](/reference/supervisor/supervisor-api/#patch-v1devicehost-config).
+* Using the device dashboard [configuration variables](/learn/manage/configuration/), you may modify the persistent logging setting.
+* Using the [configizer project](https://github.com/balena-io-playground/configizer) you may apply updates remotely via the {{ $names.cli }}. Note that this has not been thoroughly tested and should be treated with caution. Always test any changes first on a development device.
+
+Alternatively, you can always reprovision a device with an updated `config.json` file.
+
+**Note:** Editing config.json on a provisioned device is not recommended, as mistakes to this file can leave a device inaccessible. If you need to manually edit this file, see the [balenaOS masterclass instructions](https://github.com/balena-io-projects/balenaos-masterclass#12-advanced-editing-configjson-by-hand).
```

Comment:
This paragraph partially duplicates another paragraph above _("Special care should be taken...")_ Might be nicer to merge the 2 paragraphs, and perhaps emphasise a whole warning sentence in bold (if the reason to  duplicate the information was to make sure it is noticed...).

Reply:
Agreed. I'm also not sure whether we should be recommending this or not - the masterclass also has its own fair share of warnings if we do link to it :)

Reply:
> I'm also not sure whether we should be recommending this or not 

**My two cents,** I think we should, yes, because:
* The masterclasses are public and were written with the intent of sharing with end users. We share them in support interactions as well.
* In forums or paid support, users often ask about editing `config.json` and we share this kind of advice anyway, though perhaps inconsistently between support agents. Having the information in a public documentation page support effort and improves advice consistency. And if users try it and it goes wrong in some step, we will know where to focus the improvement effort.

I wonder if @shaunmulligan thinks the same way... :-)

Reply:
I agree we should share this advice, but we definitely want to really stress the point that if they make changes that makes the file unreadable, they're going to be in a position where they won't be able to access the device again. Saying that, if it's already provisioned and had connected to the VPN *and* there's another device online on the same network, we could probably salvage it. But that's not a support ticket we want to see :)

Reply:
And actually, adding to this, if people do want to mess around with it, you want to do it in a safe way: https://github.com/balena-io-playground/configizer